src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/Decider.java

```diff
@@ -91,4 +101,31 @@ public void handleNodeMuted() {
 
   @Override
   public abstract Decision operate();
+
+  /**
+   * read threshold values from rca.conf
+   *
+   * @param conf RcaConf object
+   */
+  @Override
+  public void readRcaConf(RcaConf conf) {
+    configObj = conf.getDeciderConfig();
+  }
+
+  public double getFieldDataCacheUpperBound() {
+    return configObj != null ? configObj.getFieldDataCacheUpperBound() : getDefaultFieldDataCacheUpperBound();
+  }
+
+  public double getShardRequestCacheUpperBound() {
+    return configObj != null ? configObj.getShardRequestCacheUpperBound() : getDefaultShardRequestCacheUpperBound();
+  }
+
+  public List<String> getWorkLoadPriority() {
```

Comment:
Good decision keeping them in base class. These will be required by multiple deciders.